### PR TITLE
feat(runtask): add get and patch task operation support for cstorpoolcluster

### PR DESCRIPTION
**Pull request by sonasingh46**
This PR will add get and patch operation support in CAS Teamplate for cstorpoolcluster resource.
Signed-off-by: Ashutosh Kumar <ashutosh.kumar@openebs.io>

### CODE REVIEW

File: pkg/task/task.go [hunk 1]
```diff
@@ -657,6 +661,34 @@ func (m *taskExecutor) patchOEV1alpha1SPC() (err error) {
 	return
 }
 
+// patchOEV1alpha1CSPC will patch a CSPC object in a kubernetes cluster.
+// The patch specifications as configured in the RunTask
+func (m *taskExecutor) patchOEV1alpha1CSPC() (err error) {
```
Comment (AmitKumarDas) [inline pkg/task/task.go]: For every new actions, we are making use of builder directly.
Please search for the word `builder` in this file to get references

File: pkg/client/k8s/k8s.go [hunk 2]
```diff
@@ -87,6 +87,9 @@ const (
 	// StroagePoolClaimCRKK is a K8s CR of kind StoragePool
 	StroagePoolClaimCRKK K8sKind = "StoragePoolClaim"
 
+	// CStorPoolClusterCRKK is a K8s CR of kind CStorPoolCluster
```
Comment (AmitKumarDas) [inline pkg/client/k8s/k8s.go]: Changes to this file k8s/k8s.go may not be required. Please refer to my next comments.

File: pkg/task/task.go [hunk 3]
```diff
@@ -957,6 +989,17 @@ func (m *taskExecutor) getOEV1alpha1SPC() (err error) {
 	return
 }
 
+// getOEV1alpha1CSPC() will get the CStorPoolCluster as specified in the RunTask
+func (m *taskExecutor) getOEV1alpha1CSPC() (err error) {
+	cspc, err := m.getK8sClient().GetOEV1alpha1CSPCAsRaw(m.getTaskObjectName())
```
Comment (AmitKumarDas) [inline pkg/task/task.go]: same comments. Can we make use of cspc.KubeClient()

File: pkg/task/task.go [hunk 4]
```diff
@@ -657,6 +661,34 @@ func (m *taskExecutor) patchOEV1alpha1SPC() (err error) {
 	return
 }
 
+// patchOEV1alpha1CSPC will patch a CSPC object in a kubernetes cluster.
+// The patch specifications as configured in the RunTask
+func (m *taskExecutor) patchOEV1alpha1CSPC() (err error) {
+	patch, err := asTaskPatch("patchSPC", m.runtask.Spec.Task, m.templateValues)
```
Comment (imazik) [inline pkg/task/task.go]: can we make use of new patch package here
```go
patch, err := patch.
		BuilderForRuntask("patchCSPC", m.runtask.Spec.Task, m.templateValues).
		AddCheckf(patch.IsValidType(), "given patch type is not valid").
		Build()
	if err != nil {
		return
	}
```

Comment (imazik) [reply pkg/task/task.go]:
```go
import cspc "github.com/openebs/maya/pkg/...../....."
```
```go
// patch CSPC
	p, err := cspc.
		NewKubeClient().
		Patch(m.getTaskObjectName(), patch.Type, patch.Object)
	if err != nil {
		return
	}
```

File: pkg/task/task.go (hunk 3 above)
Comment (imazik) [reply pkg/task/task.go]: something like -
```go
import cspc "github.com/openebs/maya/pkg/...../....."
```
```go
cspc, err := cspc.
		KubeClient().
		GetRaw(m.getTaskObjectName())
	if err != nil {
		return
	}
```

File: pkg/task/task.go (hunk 1 above)
Comment (AmitKumarDas) [reply pkg/task/task.go]: This can be done later if required !!!

**Pull request merged**